src/Analysis/pipeline/action_types.py

```diff
@@ -0,0 +1,453 @@
+        """
+        res = []
+        if len(array)>0: 
+            res.append(spaces*" "+name+"=[")
+            for var in array:
+                if cls._is_DTT(var):
+                    res.extend(Formater.format_parameter(var.get_settings_script(), spaces+4))
+                elif isinstance(var, BaseActionType):
+                    res.append((spaces+4)*" "+"{0},".format(var.get_instance_name()))
+                else:
+                    raise Exception(err)
+            res[-1] = res[-1][:-1]
+            res.append(spaces*" "+"],")   
+            return res
+            
+    @classmethod
+    def format_param(cls, name, var, spaces, err):
+        """
+        return lines with formated param
+        """
+        res = []
+        if cls._is_DTT(var):
+            res.extend(Formater.format_variable(name, var.get_settings_script(), spaces))
+        elif isinstance(var, BaseActionType):
+            res.append(spaces*" "+name + "={0},".format(var.get_instance_name()))
+        else:
+            raise Exception(err)
+        return res
+
+    def get_settings_script(self):    
+        """return python script, that create instance of this class"""
+        lines = []
+        lines.append("{0}_{1} = {2}(".format(self.name, str(self.id), self.__class__.__name__))
+        if len(self.inputs)==1:
+            lines.extend(self.format_param("Input", self.inputs[0], 4, "Unknown input type"))
+        elif len(self.inputs)>1:
+            lines.extend(self.format_array("Inputs", self.inputs, 4, "Unknown input type"))
+        if len(self.inputs_os)>1:
+            nonzer = False
+            var="["
+            for id in self.inputs_os:
+                var += str(id)
+                if id!=0:
+                    nonzer = True
+            if nonzer:
+                var = var[:-1]+']'
+                lines.extend(Formater.format_variable('InputOpposites', var, 4))
+        if len(self.outputs)==1:
+            lines.extend(self.format_param("Output", self.outputs[0], 4, "Unknown output type"))
+        elif len(self.outputs)>1:
+            lines.extend(self.format_array("Outputs", self.outputs, 4, "Unknown output type"))                        
+        for script in self._get_variables_script():
+            lines.extend(Formater.indent(script, 4))
+            lines[-1] += ","
+        if len(lines)==1:
+            lines[0] = lines[0]+')'
+        else:
+            lines[-1] = lines[-1][:-1]
+            lines.append(")")
+        return lines
+            
+    def get_instance_name(self):
+        return "{0}_{1}".format(self.name, str(self.id))
+        
+    def _get_variables_script(self):    
+        """
+        return array of variables as python scripts
+        each item is array of variables in format 'variable=value'
+        if value is extend to more lines, value must be closed to bracked
+        """
+        return []
+        
+    @abc.abstractmethod 
+    def _get_runner(self,  params):    
+        """
```

Comment:
Tomu nerozumím, pokud budeme potřebovat nějakou třídu pro vykonávané akce (a to asi nebude potřeba vždy), tak by naopak akce měla být parametrem konstrutoru runneru. Akce by neměla vůbec vědět v jakém kontextu je prováděna.